data-model/src/test/java/org/hypertrace/core/datamodel/shared/TraceAttributeUtilsTest.java

```diff
@@ -35,4 +37,32 @@ public void getListAttributeTest() {
         List.of("somevalue1", "somevalue2"),
         TraceAttributeUtils.getListAttribute(mockTrace, "someattribute"));
   }
+
+  @Test
+  public void getStringAttributeTest() {
+    StructuredTrace mockTrace = mock(StructuredTrace.class);
+    assertNull(TraceAttributeUtils.getStringAttribute(mockTrace, "someattribute"));
+
+    Map<String, AttributeValue> attributeMap = new HashMap<>();
+    when(mockTrace.getAttributes())
+        .thenReturn(Attributes.newBuilder().setAttributeMap(attributeMap).build());
+    assertNull(TraceAttributeUtils.getStringAttribute(mockTrace, "someattribute"));
+
+    attributeMap.put("someattribute", AttributeValue.newBuilder().setValue("somevalue").build());
+    assertEquals("somevalue", TraceAttributeUtils.getStringAttribute(mockTrace, "someattribute"));
+  }
+
+  @Test
+  public void getBooleanAttributeTest() {
+    StructuredTrace mockTrace = mock(StructuredTrace.class);
+    assertFalse(TraceAttributeUtils.getBooleanAttribute(mockTrace, "someattribute"));
+
+    Map<String, AttributeValue> attributeMap = new HashMap<>();
+    when(mockTrace.getAttributes())
+        .thenReturn(Attributes.newBuilder().setAttributeMap(attributeMap).build());
+    assertFalse(TraceAttributeUtils.getBooleanAttribute(mockTrace, "someattribute"));
+
+    attributeMap.put("someattribute", AttributeValue.newBuilder().setValue("true").build());
```

Comment:
Can you add test case for negetive case and see if its not throwing any exception?

```
attributeMap.put("someattribute", AttributeValue.newBuilder().setValue("xyz").build());
```

```
```
attributeMap.put("someattribute", AttributeValue.newBuilder().setValue("").build());
```